### PR TITLE
DOC: clarify whats required for new features see #13924

### LINKED ISSUE #13924: Contributing/pull request guidelines should clarify what's required for new features
With some frequency, we get new contributors making pull requests to NumPy for feature enhancements / API changes.

It would be great to have something to point them to that to clarify:
1. If at all possible, NumPy prefers *not* to add new APIs, but rather to leave these to more specialized packages.
2. New features (certainly new functions) generally need to be discussed on the mailing list first.

This should appear prominently in both our contributor guide in the docs and CONTRIBUTING.md:
https://github.com/numpy/numpy/blob/master/.github/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by leeyspaul**
Closes #13924. Please let me know if this is no longer needed!

### CODE REVIEW

File: doc/source/dev/index.rst [hunk 1]
```diff
@@ -173,6 +173,7 @@ Guidelines
 * No changes are ever committed without review and approval by a core
   team member.Please ask politely on the PR or on the `mailing list`_ if you
   get no response to your pull request within a week.
+* If you would like to create a new feature please get alignment first by discussing the feature in the `mailing list`_
```
Comment (rossbar) [inline doc/source/dev/index.rst:176]:
```suggestion
```
The addition of this line here might be a bit confusing, as these are general guidelines but the mailing list suggestion is only for new feature proposals. We don't want to make new users feel like they have to ping the mailing list before contributing anything.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -29,3 +29,10 @@ If you are writing new C code, please follow the style described in
 Suggested ways to work on your development version (compile and run
 the tests without interfering with system packages) are described in
 ``doc/source/dev/development_environment.rst``.
+
+### A note on feature enhancements/API changes
+
+1. If you are looking to enhance features or make any API changes please note that NumPy
+prefers not to add new APIs but instead leave these to more specialized packages.
+2. If you would like to add a new feature please make sure you raise the feature in our
+mailing list first to get alignment. You can sign up for the mailing list [here](https://mail.python.org/mailman/listinfo/numpy-discussion).
```
Comment (rossbar) [inline .github/CONTRIBUTING.md:38]:
```suggestion
If you are interested in adding a new feature to NumPy, consider
submitting your feature proposal to the `mailing_list`_, which is
the preferred forum for discussing new features and API changes.
```
Just a suggestion to make the guidelines seem less formulaic - feel free to ignore. (n.b. the mailing list link might need to be fixed in this suggestion)

**Review by rossbar — COMMENTED** [review]
It wouldn't hurt having a note about the mailing list for first time contributors, though we definitely want to make clear that it's just a suggestion and not *required* as new contributors may be unfamiliar/uncomfortable with the mailing list.

**Comment by leeyspaul** [comment]
> It wouldn't hurt having a note about the mailing list for first time contributors, though we definitely want to make clear that it's just a suggestion and not _required_ as new contributors may be unfamiliar/uncomfortable with the mailing list.

I like both your ideas! Thank you for them :). I went ahead and committed the suggested changes. Do let me know if there is anything else!

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -29,3 +29,9 @@ If you are writing new C code, please follow the style described in
 Suggested ways to work on your development version (compile and run
 the tests without interfering with system packages) are described in
 ``doc/source/dev/development_environment.rst``.
+
+### A note on feature enhancements/API changes
+
+If you are interested in adding a new feature to NumPy, consider
+submitting your feature proposal to the `mailing_list`_, which is
+the preferred forum for discussing new features and API changes.
```
Comment (rossbar) [inline .github/CONTRIBUTING.md:37]:
```suggestion
If you are interested in adding a new feature to NumPy, consider
submitting your feature proposal to the [mailing list][mail], which is
the preferred forum for discussing new features and API changes.

[mail]: https://mail.python.org/mailman/listinfo/numpy-discussion
```

**Review by rossbar — APPROVED** [review]
LGTM - I took the liberty of fixing my suggested mailing list link (I had originally used rST syntax, not md). Thanks for the improvement @leeyspaul !

**Comment by rossbar** [comment]
For reviewers: this isn't part of the documentation, and is usually only seen by first-time contributors to the project on GitHub. Here's a link to the rendered contribution message: https://github.com/leeyspaul/numpy/blob/clarify-new-feature-requirements-in-contributing-docs/.github/CONTRIBUTING.md

**Comment by melissawm** [comment]
Thanks, @leeyspaul and @rossbar !

**Review by melissawm — APPROVED** [review]

**Comment by mattip** [comment]
Should we mention something about "please paste text formatted as code rather than screenshots" and something about the COC?

**Comment by rossbar** [comment]
> Should we mention something about "please paste text formatted as code rather than screenshots" and something about the COC?

I think the CONTRIBUTING.md doc is only linked to when creating PRs, not issues, so the part about text formatted code would be a bit out of place. A link to the code of conduct could certainly be added, though I'm not sure how prevalent CoC problems are with first-time contributor PRs.

**Comment by mattip** [comment]
Thanks @leeyspaul and @rossbar

**Pull request merged**